Refactor

nicknisi opened the pull request:
Resolves issue #26 

**Reviewers**:
- [ ] @jgchristopher
- [x] @andymoon
- [x] @doug-martin

This refactor separates thumbs into multiple, testable files. A new build step using grunt-preprocess will concat all the files into a single file, for distribution and minification. I also added a LICENSE file, and added the license comments to the top of the file
## Installing Dependencies

`npm install && bower install`

> This refactor also updates this package for bower 0.9, of which the biggest change is the renaming of component.json to bower.json.
## Testing

Test the files with jasmine in phantom by running `grunt test`
## Building

To build the concatenated and minified versions of thumbs, run `grunt build`, or simply run `grunt`

nicknisi committed:
Separating Thumbs into multiple files

+ created Thumbs namespace
    + thumbs namespace still exists for backwards compatibility
+ updated Gruntfile Jasmine specs for new Files

nicknisi committed:
Got TemplateView and Class added/tested

getting build to work'

nicknisi committed:
added andy's changes to new structure

nicknisi committed:
testing with backbone 1.0

working on tests

Added core.spec.js tests

core.spec.js mixes the old thumbsSpec.js and _superSpec.js files

adding class.spec.js

got delegate specs running

Got elFinder specs running

got format specs running

adding id.spec and made it pass

getting view specs to work

adding template specs

got monitor specs working

got subview specs working

nicknisi committed:
added Thumbs.noConflict and spec for it

Added AMD support

nicknisi committed:
removing old jasmine runner and libs

Jamsine is now installed via Bower and run via grunt task

nicknisi committed:
Updating to bower 0.9

nicknisi committed:
Added LICENSE

Added MIT license file

fixing build, cleaning up grunt, adding license

nicknisi committed:
added jshintignore

nicknisi committed:
Renaming src/ to lib/

doug-martin commented on lib/thumbs.core.js:
```diff
@@ -0,0 +1,155 @@
+    Backbone.Model.extend = Backbone.Collection.extend = Backbone.View.extend = Backbone.History.extend = Backbone.Router.extend = Thumbs.extend;
+
+    Thumbs._super = {
+        _super: (function () {
```
It seems we have _super defined twice both in helpers below and here

nicknisi replied:
my bad. I forgot to delete those two files. Fixed.

nicknisi committed:
Removing unnecessary files and cleaning up

_super was being defined three times

andymoon commented on thumbs.js:
```diff
@@ -118,821 +90,697 @@
-    }
 
+    root.thumbs = Thumbs;
 
```
Will this still work if Backbone isn't already loaded?  I think it used to check if it was globally declared and if not it would do a require for Backbone and underscore.

nicknisi committed:
removing global thumbs declaration

nicknisi committed:
renaming lib to src

this appears to be the pattern i see mostly on other projects: lib
contains the built versions of the code and src is the unbuilt/dev
versions.

nicknisi committed:
Removing unnecessary args to IIFEs in Files

nicknisi committed:
Placing require/amd at bottom

doug-martin commented on src/thumbs.class.js:
```diff
@@ -0,0 +1,15 @@
+Thumbs.Class = (function () {
```
We talked about this and you said you would remove the iife

nicknisi committed:
Fixing issues with amd/commonjs loader

+ added defineThumbs function back in

nicknisi committed:
changing instance of backbone, underscore loaded to root. (this)
